lib/bloc/blocs/auth_provider.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:structure_flutter/core/common/enums/authenticate.dart';
+import 'file:///G:/Project/jobchat/Structure_Flutter/lib/widgets/snackbar_service.dart';
+import 'package:structure_flutter/core/extension/navigation_extension.dart';
+import 'package:structure_flutter/repositories/account_repository.dart';
+
+class AuthProvider extends ChangeNotifier {
+  User user;
+  AuthStatus status;
+
+  FirebaseAuth auth;
+
+  static AuthProvider instance = AuthProvider();
+
+  AuthProvider() {
+    auth = FirebaseAuth.instance;
+    _checkCurrentUserIsAuthenticated();
+  }
+
+  void _autoLogin() async {
+    if (user != null) {
+      await AccountRepositoryImpl.instance.updateUserLastSeenTime(user.uid);
+      return NavigationExtension.instance.navigateToReplacement("home");
+    }
+  }
+
+  void _checkCurrentUserIsAuthenticated() async {
+    user = await auth.currentUser;
+    if (user != null) {
+      notifyListeners();
+      await _autoLogin();
+    }
+  }
+
+  void loginUserWithEmailAndPassword(String _email, String _password) async {
+    status = AuthStatus.Authenticating;
+    notifyListeners();
+    try {
+      UserCredential _userCredential = await auth.signInWithEmailAndPassword(
+          email: "$_email", password: "$_password");
```

Comment:
```_email``` and ```_password``` variables are already ```String``` type

Reply:
i already checked it but it not working when using String type in this case